### PR TITLE
Add tests for the filter tree

### PR DESCRIPTION
**Pull request by tahini**
Signed-off-by: Geneviève Bastien <gbastien+lttng@versatic.net>

### CODE REVIEW

**Comment by tahini** [comment]
The tests are failing for now, and it's a legit bug!! Will fix later, but it gives the example tests

**Comment by tahini** [comment]
This is ready for review. @chertyb, I added tests for the filter box too

**Comment by tahini** [comment]
@chertyb can you try again now. Strangely, I _upgraded_ the ts-jest package and now it works here...

File: viewer-prototype/jest.config.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    "testRegex": "(/__tests__/.*|(\\.|/)(test|spec))\\.(jsx?|tsx?)$",
+    "moduleFileExtensions": [
+        "ts",
+        "tsx",
+        "js",
+        "jsx",
+        "json",
+        "node"
+    ]
+}
```
Comment (bhufmann) [inline viewer-prototype/jest.config.json:20]: Github diff shows warning "No newline at end of file"

File: viewer-prototype/src/browser/style/output-components-style.css [hunk 2]
```diff
@@ -62,6 +62,29 @@ canvas {
...
+}
+
+.timegraph-tree tr {
+    line-height: 18px;  /* arbitrary, it works [on my machine], but it should match line height in timeline-chart */
+    position: relative;
+    white-space: nowrap;
+    top: 50%;
+    padding: 0 0;
+}
+
```
Comment (bhufmann) [inline viewer-prototype/src/browser/style/output-components-style.css:87]: shouldn't this be part of the parent PR?

File: viewer-prototype/src/browser/trace-viewer/components/timegraph-output-component.tsx [hunk 3]
```diff
@@ -130,12 +130,15 @@ export class TimegraphOutputComponent extends AbstractTreeOutputComponent<Timegr
     }
 
     renderTree(): React.ReactNode {
+        // TODO Show header, when we can have entries in-line with timeline-chart
```
Comment (bhufmann) [inline viewer-prototype/src/browser/trace-viewer/components/timegraph-output-component.tsx:133]: Shouldn't it this change be of the parent PR?

File: viewer-prototype/src/browser/trace-viewer/components/timegraph-output-component.tsx [hunk 4]
```diff
@@ -130,12 +130,15 @@ export class TimegraphOutputComponent extends AbstractTreeOutputComponent<Timegr
...
         return <EntryTree
             collapsedNodes={this.state.collapsedNodes}
             showFilter={false}
             entries={this.state.timegraphTree}
             showCheckboxes={false}
             onToggleCollapse={this.onToggleCollapse}
+            showHeader={false}
+            className="timegraph-tree"
```
Comment (bhufmann) [inline viewer-prototype/src/browser/trace-viewer/components/timegraph-output-component.tsx:141]: Shouldn't it this change be of the parent PR?

**Review by bhufmann — COMMENTED** [review]
Pretty cool. It includes another patch (PR #116). If the other patch is merged then this one can be rebased.

**Comment by bhufmann** [comment]
Just for curiosity, are there any tools to create test case skeleton? Or do we need to write them by hand?

**Comment by tahini** [comment]
> @tahini, just for curiosity, are there any tools to create test case skeleton? Or do we need to write them by hand?

There exists some tools out there to build test cases in jest. I'm not familiar enough with them though, I may look into it soon and see if we could use some of them. For now, we need to write them by hand. Also, for another project, I also played with enzyme which makes some kind of tests easier. This patch is just an example to bootstrap tests in our codebase. I'll add some more soon, here and everywhere!!!

**Comment by bhufmann** [comment]
> > @tahini, just for curiosity, are there any tools to create test case skeleton? Or do we need to write them by hand?
> 
> There exists some tools out there to build test cases in jest. I'm not familiar enough with them though, I may look into it soon and see if we could use some of them. For now, we need to write them by hand. Also, for another project, I also played with enzyme which makes some kind of tests easier. This patch is just an example to bootstrap tests in our codebase. I'll add some more soon, here and everywhere!!!

Thanks for the info.

**Comment by tahini** [comment]
This PR is now ready for review, all pre-requisites are in

**Review by MatthewKhouzam — APPROVED** [review]
Looks good to me, I am still weirded out by the lack of copyright headers.

**Comment by tahini** [comment]
Rebased on master, I'll push this and let the era of unit tests in the trace-extension begin!

**Pull request merged**